src/consensus/pbft/pbft.h

```diff
@@ -594,21 +632,40 @@ namespace pbft
     static void recv_authenticated_msg_cb(
       std::unique_ptr<enclave::Tmsg<RecvAuthenticatedMsg>> msg)
     {
-      try
+      if (msg->data.should_decrypt)
```

Comment:
why are we using `msg->data.should_decrypt` to determine whether a message is encrypted or not, instead of utilizing the `encrypted_pbft_message` message type?

Reply:
I am, that is how this property I set. I might be misunderstanding your comment.

Reply:
ah yes I didn't notice the threading callback in the PR diff